docs/adding-a-new-delta.md

```diff
@@ -27,77 +25,12 @@ paths:
-as a guide.
-
-- The KafkaService will be needed to send your data onto a given Kafka topic.
-- The helper struct will be needed to extract the contextId from your request and to marshal your request into a string 
-ready for sending to Kafka.
-- The CHValidator will be used to validate your request against your newly created api-spec.
-- The config struct will be used to load all needed configs.
-
-```go
-type ExampleHandler struct {
-	kSvc services.KafkaService
-	h    helpers.Helper
-	chv  validation.CHValidator
-	cfg  *config.Config
-}
-
-func NewExampleDeltaHandler(kSvc services.KafkaService, h helpers.Helper, chv validation.CHValidator, cfg *config.Config) *ExampleDeltaHandler {
-	return &ExampleDeltaHandler{kSvc: kSvc, h: h, chv: chv, cfg: cfg}
-}
-
-
-func (eh *ExampleDeltaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	contextId := eh.h.GetRequestIdFromHeader(r)
-    
-    log.InfoC(contextId, fmt.Sprintf("Using the openAPI spec: "), log.Data{config.OpenApiSpecKey: eh.cfg.OpenApiSpec})
-
-    // Validate against the openAPI 3 spec before progressing any further.
-    errValidation, err := eh.chv.ValidateRequestAgainstOpenApiSpec(r, eh.cfg.OpenApiSpec, contextId)
-    if err != nil {
-        w.WriteHeader(http.StatusInternalServerError)
-        log.ErrorC(contextId, err, log.Data{config.MessageKey: "error occurred while trying to validate request"})
-        return
-    } else if errValidation != nil {
-        w.WriteHeader(http.StatusBadRequest)
-        _, err = w.Write(errValidation)
-        if err != nil {
-            log.ErrorC(contextId, err, log.Data{config.MessageKey: "error occurred while trying to write response"})
-        }
-
-        return
-    }
-
-    // Get request body and marshal into a string, ready for publishing.
-    data, err := eh.h.GetDataFromRequest(r, contextId)
-    if err != nil {
-        w.WriteHeader(http.StatusInternalServerError)
-        return
-    }
-
-    // Send data string to Kafka service for publishing.
-    if err := eh.kSvc.SendMessage(eh.cfg.ExampleDeltaTopic, data, contextId); err != nil {
-        log.ErrorC(contextId, err, log.Data{config.TopicKey: eh.cfg.ExampleDeltaTopic, config.MessageKey: "error sending the message to the given kafka topic"})
-        w.WriteHeader(http.StatusInternalServerError)
-
-        return
-    }
-
-    w.WriteHeader(http.StatusOK)
-}
 ```
 
-## 4. Exposing your new delta endpoint
+## 3. Exposing your new delta endpoint
 The final step of adding a new delta is to register your new route within the `register.go` file.
 ```go
-appRouter.HandleFunc("/delta/example-delta", NewExampleDeltaHandler(kSvc, h, chv, cfg).ServeHTTP).Methods(http.MethodPost).Name("example-delta")
+appRouter.HandleFunc("/delta/example-delta", NewDeltaHandler(kSvc, h, chv, cfg, true, cfg.ExampleDeltaTopic).ServeHTTP).Methods(http.MethodPost).Name("example-delta")
```

Comment:
Here, might be worth explaining they can add a validate endpoint as well and what the boolean flag is for

Reply:
Added this, good point!